### DISCUSSION TITLE
Models have JSON output

### DISCUSSION THREAD

**Discussion by blairliikala** [ExpressionEngine/ExpressionEngine#2862] [Feature Requests]
Models should include a method to output JSON for better communication with front-ends and external services.  The goal is better developer experience, and less boilerplate needed.  While it is possible to get to JSON from a model, it could be more elegant.

Example:
```php
$string = ee('Model')->get('stuff')->first()->json('pretty');
```

Currently it looks something like this:
```php
$models = ee('Model')->get('stuff')...etc...->all();
$models_array = $models->toArray();
$models_json = json_encode($models_array);
```
Or

```php
$model = ee('Model')->get('stuff')...etc...->first();
$model_array[] = $model;
$model_json = json_encode($model_array);
```

[x] I am capable and would like to work on implementation of this feature if it is considered.

**Comment by intoeetive** [comment]
@blairliikala have `json` method added natively wouldn't do anything bad certainly :)
If all you want to do its to add wrapper for `json_encode` then I'd say go ahead for it - but if you have something more complex in mind, then we might need to have a discussion first

**Comment by blairliikala** [comment]
Mostly a wrapper, yes.  If this worked then it wouldn't be as annoying:

```php
$models = ee('Model')->get('File')->all(); // or ->first()
$string = json_encode($models);

// Outputs {}
```
But this outputs an empty object.  You have to use the `toArray()` and that took me [an embarrassing] awhile to figure out.
